doc: Adjusting new sharding configuration migration guide with auto-scaling considerations + examples (#2879)
mongodb/terraform-provider-mongodbatlas: examples/mongodbatlas_advanced_cluster/auto-scaling-per-shard/README.md

# MongoDB Atlas Provider -- Sharded Cluster with Independent Shard Auto-scaling

> This example creates a Sharded Cluster with 2 shards defining electable and analytics nodes. Compute auto-scaling is enabled for both `electable_specs` and `analytics_specs`, while also leveraging the [New Sharding Configuration](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema) by defining each shard with its individual `replication_specs`. This enables scaling of each shard to be independent. Please reference [Use Auto-Scaling Per Shard](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema#use-auto-scaling-per-shard) section for more details.

Reviewer: missing article before cross reference
```suggestion
This example creates a Sharded Cluster with 2 shards defining electable and analytics nodes. Compute auto-scaling is enabled for both `electable_specs` and `analytics_specs`, while also leveraging the [New Sharding Configuration](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema) by defining each shard with its individual `replication_specs`. This enables scaling of each shard to be independent. Please reference the [Use Auto-Scaling Per Shard](https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs/guides/advanced-cluster-new-sharding-schema#use-auto-scaling-per-shard) section for more details.
```

## Dependencies

* Terraform MongoDB Atlas Provider v1.23.0
* A MongoDB Atlas account 

```

> Terraform >= 0.13

Reviewer 2: Should this be updated to maybe lowest supported TF version? Just noticed we have this across all advanced_cluster examples
Or at least align with the example which uses `required_version = ">= 1.0"`

Author: Good catch, would align with other examples and simply not mention it. In terms of `required_version = ">= 1.0"` also thinking we could remove them from our examples so we have single point of reference for terraform version support. WDYT? I can create a ticket for this.

Reviewer 2: Yes, I agree we should remove this altogether and have a single point of reference

+ provider registry.terraform.io/terraform-providers/mongodbatlas v1.23.0
```

## Usage

> **1\. Ensure your MongoDB Atlas credentials are set up.**

Reviewer: active voice

```suggestion
**1\. If you haven't already, set up your MongoDB Atlas credentials.**
```

This can be done using environment variables:

```bash
export MONGODB_ATLAS_PUBLIC_KEY="xxxx"
export MONGODB_ATLAS_PRIVATE_KEY="xxxx"
```

... or follow as in the `variables.tf` file and create **terraform.tfvars** file with all the variable values, ex:
```
public_key           = "<MONGODB_ATLAS_PUBLIC_KEY>"
private_key          = "<MONGODB_ATLAS_PRIVATE_KEY>"
atlas_org_id         = "<MONGODB_ATLAS_ORG_ID>"
```

> ... or use [AWS Secrets Manager](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/website/docs/index.html.markdown#aws-secrets-manager)

Reviewer: ```suggestion
Alternatively, you can use [AWS Secrets Manager](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/website/docs/index.html.markdown#aws-secrets-manager).
```

**2\. Review the Terraform plan.**

Execute the below command and ensure you are happy with the plan.

``` bash
$ terraform plan
```
This project currently supports the below deployments:

- An Atlas Project
- A Sharded Cluster with independent shards with varying cluster tiers

> **3\. Execute the Terraform apply.**

Reviewer: ```suggestion
**3\. Apply your changes.**
```
